datastore/gcloud/aio/datastore/value.py

```diff
@@ -74,6 +77,10 @@ def to_repr(self) -> Dict[str, Any]:
     def _infer_type(self, value: Any) -> TypeName:
         odd_strings = {'newstr', 'unicode'}
         kind = str if value.__class__.__name__ in odd_strings else type(value)
+        # We should instead use Array as the key in TYPES directly, but that
```

Comment:
My [comment on your other PR](https://github.com/talkiq/gcloud-aio/pull/139#discussion_r377963227) actually gave me a thought for a way to avoid this hack -- rather than adding `Array` to `TYPES`, adding it directly in the `_get_supported_types()` method will allow you to nuke this hack and avoid the import cycle issue you ran into.